modules/phase_field/src/postprocessors/GrainTracker.C

```diff
@@ -143,20 +147,84 @@ GrainTracker::finalize()
 
   expandHalos();
 
-  FeatureFloodCount::communicateAndMerge();
+  // Build up the grain map on the root processor
+  communicateAndMerge();
+
+  /**
+   * Track Grains
+   */
+  Moose::perf_log.push("trackGrains()", "GrainTracker");
+  std::vector<unsigned int> new_grain_indices;
+
+  // Track grains (only on the root processor)
+  if (_is_master)
+    trackGrains(new_grain_indices);
+
+  if (!new_grain_indices.empty())
```

Comment:
Holy crap! Then that's a bug! Nice catch. I know I have a test for this but we don't run the module tests in parallel yet. This will definitely hang if a new grain is created.